src/components/ProductRecommendations/index.jsx

```diff
@@ -0,0 +1,109 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import Kvass from '@kvass.ai/core-sdk';
+import T from 'src/utils/i18n';
+import ProductImage from 'src/components/ProductImage';
+import Spinner from 'src/components/Spinner';
+import Actions from 'src/reflux/Actions';
+import utils from 'src/utils';
+
+class ProductRecommendations extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: false,
+      recommendations: [],
+      visibleRecIndex: 3, // endIndex of what we are showing from array
+      size: 9 // no pagination on recommendations
+    };
+    this.kvass = new Kvass();
+  }
+
+  componentDidMount() {
+    const body = {model_type: this.props.modelType, source_id: this.props.sourceId, source: this.props.source, destination: this.props.destination, size: this.state.size};
+    this.setState({isLoading: true});
+    this.kvass.aiModel().getRecommendations({body, urlParams: {expand: 'response.object'}}, (error, recommendations, res) => {
+      if (error) {
+        this.setState({isLoading: false, recommendations: []});
+        return;
+      }
+      this.setState({isLoading: false, recommendations: utils.parseRecommendations(recommendations)});
+    });
+  }
+
+  onNavigateRecommendations(direction) {
+    if (!direction) return;
+    let newVisibleIndex = this.state.visibleRecIndex;
+    this.setState({isLoading: true});
+
+    if (direction === 'back') {
+      newVisibleIndex -= 3;
+    } else if (direction === 'forward') {
+      newVisibleIndex += 3;
+    }
+
+    setTimeout(() => {
+      this.setState({
+        visibleRecIndex: (newVisibleIndex >= 3 && newVisibleIndex <= this.state.size) ? newVisibleIndex : 3,
+        isLoading: false
+      });
+    }, 500);
+  }
+
+  onProductClick(product) {
+    Actions.onSelectProduct(product);
+    Actions.onNavigateTo(7); // Navigate to product page
+  }
+
+  renderRecommendationItem(product) {
+    return (
+      <li key={product._id.$oid} className="recommendation__item" onClick={() => this.onProductClick(product)}>
+        <ProductImage imageUrl={product.image_url} apiKey={this.kvass.apiKey} endpoint={this.kvass.endpoint} />
```

Comment:
This feels kinda strange. Is it possible to initialize kvass in the productImage, so we don't have to pass the endpoint and the key?

Reply:
ProductImage is a stateless function component, it just uses the data it gets in. It gets called quite a lot since it doesn't have react lifecycle method

Reply:
Yeah. Is it supposed to be reusable? Maybe its better to make it stateless react component that initializes kvass, thats in case we consider it reusable.